alica_engine/src/engine/allocationauthority/AuthorityManager.cpp

```diff
@@ -47,22 +47,22 @@ void AuthorityManager::handleIncomingAuthorityMessage(const AllocationAuthorityI
 {
     AlicaTime now = _engine->getAlicaClock().now();
     TeamManager& tm = _engine->editTeamManager();
-    if (tm.isAgentIgnored(aai.senderID)) {
+    if (aai.senderID == _localAgentID || tm.isAgentIgnored(aai.senderID)) {
         return;
     }
-    if (*(aai.senderID) != *_localAgentID) {
-        tm.setTimeLastMsgReceived(aai.senderID, now);
-        if (*(aai.senderID) > *_localAgentID) {
-            // notify TO that evidence about other robots is available
-            for (EntryPointRobots epr : aai.entryPointRobots) {
-                for (AgentIDConstPtr rid : epr.robots) {
-                    if (*rid != *_localAgentID) {
-                        tm.setTimeLastMsgReceived(rid, now);
-                    }
+
+    tm.setTimeLastMsgReceived(aai.senderID, now);
+    if (_localAgentID < aai.senderID) {
```

Comment:
This won't work.
less-than over AgentIdConstPtr is not guaranteed to be consistent between different agents. This is required here though.

Maybe we should add a comment.

Reply:
Looks like I am missing something? "operator <" is invoking the same thing on AgentId. When would they not be consistent? And also I could not decipher this special case of setting the time.

Reply:
Ah, that's what I get for reviewing without ide. This is part of the bully algorithm, where the total order over ids is used to establish agreement.